WeakEvent/WeakEventSource.cs

```diff
@@ -23,7 +23,13 @@ public void Raise(object sender, TEventArgs e)
         {
             lock (_handlers)
             {
-                _handlers.RemoveAll(h => !h.Invoke(sender, e));
+                var failedHandlers = _handlers
+                    .ToArray()
+                    .Where(h => !h.Invoke(sender, e))
+                    .ToArray();
```

Comment:
The second `ToArray` isn't necessary.
In fact, the code would probably be more readable if the call `Invoke` was made in the loop. In retrospect, the one-liner with `RemoveAll` was probably not a great idea...

Reply:
Both `ToArray` was required to make my project work. I'll try add until tests soon to illustrate. In the meantime, if you want to check out the `netstandard2-weakevents` branch on https://github.com/igitur/ClosedXML you van see the problem.

Reply:
Are you sure? The second `ToArray` just enumerates the query eagerly, which is going to be done in the loop anyway.

> if you want to check out the #netstandard2-weakevent branch

I'm not sure what I'm supposed to be looking at in this branch...

Reply:
You were right. The extra `ToArray` isn't necessary. Friday nights aren't the best for coding. :-P